yt_dlp/YoutubeDL.py

```diff
@@ -3907,58 +3903,110 @@ def get_encoding(stream):
                     'See https://yt-dl.org/update if you need help updating.' %
                     latest_version)
 
-    def _setup_opener(self):
-        if hasattr(self, '_opener'):
-            return
-        timeout_val = self.params.get('socket_timeout')
-        self._socket_timeout = 20 if timeout_val is None else float(timeout_val)
-
-        opts_cookiesfrombrowser = self.params.get('cookiesfrombrowser')
-        opts_cookiefile = self.params.get('cookiefile')
+    @functools.cached_property
+    def proxies(self) -> dict:
```

Comment:
do we want this to be public?

In the future we may want to allow specifying a proxy map like this in the ydl params. e.g. `ydl.params['proxies']`

Reply:
should be fine, this is what everything should access instead of ydl.params['proxy']